### PR TITLE
Allow collecting coverage data from in-tree tests

### PR DESCRIPTION
**Pull request by liuw**
This requires changing the test scripts a bit.

Add some notes on how to do that in coverage.md.

### CODE REVIEW

**Comment by jinankjain** [comment]
I think since we know have a way to collect coverage data, would it make sense to run a special coverage build as part of PR pipeline and publish the produced htmls files on some Azure storage blob account.

File: docs/coverage.md [hunk 1]
```diff
@@ -53,3 +53,29 @@ grcov . --binary-path ./target/x86_64-unknown-linux-gnu/release -s . -t html --b
...
+```shell
+# Get a shell
+./scripts/dev_cli.sh shell
+
+# Install llvm-tools-preview for llvm-profdata
+rustup component add llvm-tools-preview
+# Merge data files by using the following command
+find . -name '*.profraw' -exec `rustc --print sysroot`/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata merge -sparse {} -o coverage.profdata \;
+
+# As of writing, the container has Rust 1.67.1. It is too old for grcov.
```
Comment (jinankjain) [inline docs/coverage.md:76]: Should we also upgrade the container to the latest stable maybe?

Comment (liuw) [reply docs/coverage.md:76]: I'm a bit hesitant. Anyone who wants to process the data still needs to install llvm-tools-preview by hand.

If this were needed for automated systems I would be more inclined to upgrade the container.

**Review by rbradford — APPROVED** [review]

File: scripts/dev_cli.sh [hunk 2]
```diff
@@ -396,6 +396,13 @@ cmd_tests() {
     process_volumes_args
     target="$(uname -m)-unknown-linux-${libc}"
 
+    rustflags="$RUSTFLAGS"
+    target_cc=""
+    if [ "$(uname -m)" = "aarch64" ] && [ "$libc" = "musl" ]; then
+        rustflags="$rustflags -C link-arg=-lgcc -C link_arg=-specs -C link_arg=/usr/lib/aarch64-linux-musl/musl-gcc.specs"
+        target_cc="musl-gcc"
```
Comment (likebreath) [inline scripts/dev_cli.sh:403]: @liuw Just want to point out that `target_cc` is only set for `aarch64` + `musl`. This is different from previous behavior, where we always set `target_cc="musl-gcc"` as long as we are building with `musl` libc. No sure if this is intended.

Comment (liuw) [reply scripts/dev_cli.sh:403]: I noticed that. I _think_ it is for cross-compilation. When we're not cross-compiling, setting it to empty has no effect. That should be the default value anyway.

**Pull request merged**